src/queue.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+
+export default class Queue {
+  constructor(initialValues=[]) {
+    this.elements = initialValues
+    this.first = initialValues[0]
+    this.last = initialValues.length
```

Comment:
Why do you need a reference to the length of the initial values? 

`this.elements` is an array, which has a built in `length` property.